src/bytearray.rs

```diff
@@ -0,0 +1,79 @@
+use rand::prelude::*;
+#[derive(Debug)]
+pub struct ByteArray {
+    /// Array of bytes
+    pub data: Vec<u8>,
+}
+
+impl ByteArray {
+    pub fn random(len: usize) -> Self {
+        let mut data = vec![0; len];
+        let mut rng = rand::thread_rng();
+        rng.fill_bytes(&mut data);
+
+        Self { data }
+    }
+
+    pub fn append(&self, other: &Self) -> Self {
+        let mut data = self.data.clone();
+        data.extend(other.data.iter().cloned());
+        Self { data }
+    }
+
+    pub fn concat(items: &[&Self]) -> Self {
+        if items.len() == 0 {
+            return Self { data: vec![] };
+        }
+
+        let mut data = items.first().unwrap().data.clone();
+        for item in items.iter().skip(1) {
+            data.extend(item.data.iter().cloned());
```

Comment:
I think you should prefer using [`Vec::extend_from_slice`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.extend_from_slice) here.